### ISSUE TITLE
Please add support for intel pytorch for xpu (intel iGPU)

### ISSUE THREAD

**Issue by LIJUCHACKO** [ultralytics/ultralytics#21638]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

I was able to make some modifications in the code for inference and training on cpu(intel iGPU). I have done finetuning with custom dataset on yolo11s-seg.pt and yolo11l-seg.pt models.
yolo11s used around 32 GB of RAM
yollo11l  used around 59 GB of RAM
If training is done on cpu RAM required is less , almost half. After every epoch there was slightly increase in RAM. After interrupting and on resuming the training RAM usage is less than on previous run.
These are the major changes I have 
- replacing -> intel.cuda... with intel.xpu...
- In function that return amp . Return value should be false for xpu

### Use case

SYSTEM USED FOR TRAINING 
- Intel ultra core 7 155h processor 
-  64 GB ram
- UBUNTU 24.04LTS OS

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @LIJUCHACKO, thank you for your interest in Ultralytics 🚀! This is an automated assistant response — an Ultralytics engineer will review and follow up soon.

If this is a 🐛 Bug Report, please provide a minimum reproducible example (MRE) to help us debug it: https://docs.ultralytics.com/help/minimum-reproducible-example/. For feature requests like Intel XPU support, sharing a concise code diff or reproducible snippet demonstrating your changes (e.g., intel.cuda → intel.xpu replacements and AMP handling) will greatly help the team evaluate feasibility and scope.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our Tips for Best Training Results: https://docs.ultralytics.com/guides/model-training-tips/.

Join the Ultralytics community where it suits you best. For real-time chat, head to Discord 🎧: https://discord.com/invite/ultralytics. Prefer in-depth discussions? Check out Discourse: https://community.ultralytics.com/. Or dive into threads on our Subreddit to share knowledge with the community: https://www.reddit.com/r/Ultralytics/.

## Upgrade

Upgrade to the latest ultralytics package including all requirements in a Python>=3.8 environment with PyTorch>=1.8 to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

Requirements: https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml  
Python: https://www.python.org/  
PyTorch: https://pytorch.org/get-started/locally/

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):

- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See GCP Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/
- Amazon Deep Learning AMI. See AWS Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/
- Docker Image. See Docker Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/ <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all Ultralytics CI tests are currently passing. CI tests verify correct operation of all YOLO Modes and Tasks on macOS, Windows, and Ubuntu every 24 hours and on every commit.

To help us evaluate Intel XPU support specifically:
-1) Share the exact ultralytics and PyTorch versions, and your training/inference commands  
-2) Provide a minimal code snippet showing the intel.xpu device selection and AMP change you made  
-3) Include memory usage logs or a short log excerpt illustrating the RAM growth per epoch and after resume  
-4) Note which models you tested (e.g., YOLO11s-seg, YOLO11l-seg) and batch/image sizes

Thanks again for the detailed context and willingness to explore Intel iGPU support 🙌

**Comment by tommchugh1** [comment]
Hello sir,\
You are a god among men. Such a niche issue but you have helped me immensely, thankyou for your efforts.

**Comment by glenn-jocher** [comment]
Thanks @LIJUCHACKO — great findings. @tommchugh1 glad it helped, and shout‑out to the community and Ultralytics team for making YOLO11 adaptable.
